lib/dl_core_testing/dl_core_testing/testcases/adapter.py

```diff
@@ -0,0 +1,87 @@
+import abc
+from typing import (
+    ClassVar,
+    Generic,
+    Type,
+    TypeVar,
+)
+
+import pytest
+
+from dl_api_commons.base_models import RequestContextInfo
+from dl_core import exc
+from dl_core.connection_executors import AsyncConnExecutorBase
+from dl_core.connection_executors.adapters.async_adapters_base import AsyncDirectDBAdapter
+from dl_core.connection_executors.models.connection_target_dto_base import BaseSQLConnTargetDTO
+from dl_core.connection_executors.models.db_adapter_data import DBAdapterQuery
+from dl_core.connection_executors.models.scoped_rci import DBAdapterScopedRCI
+from dl_core_testing.testcases.connection_executor import BaseConnectionExecutorTestClass
+
+
+_TARGET_DTO_TV = TypeVar("_TARGET_DTO_TV", bound=BaseSQLConnTargetDTO)
+
+
+class BaseAsyncAdapterTestClass(BaseConnectionExecutorTestClass, Generic[_TARGET_DTO_TV], metaclass=abc.ABCMeta):
+    """
+    Most of the testable adapter behaviour is covered by conn executor tests
+    Here are the tests, that are just easier to implement at the adapter level
+    """
+
+    ASYNC_ADAPTER_CLS: ClassVar[Type[AsyncDirectDBAdapter]]  # TODO add tests for other adapters
+
+    @pytest.fixture
+    async def target_conn_dto(self, async_connection_executor: AsyncConnExecutorBase) -> _TARGET_DTO_TV:
+        return next(iter(await async_connection_executor._make_target_conn_dto_pool()))  # noqa
```

Comment:
can we avoid using `noqa` here? maybe just pick a first item from list instead of `next(iter())`?

Reply:
`next(iter())` is not the problem here, the problems are:
- this convenient method `_make_target_conn_dto_pool` is implemented only for `DefaultSqlAlchemyConnExecutor`
- it is a protected method

These tests are very raw and targeted on a specific subset of connectors. We don't have profound adapter tests at the moment and I was not planning to [implement](https://github.com/datalens-tech/datalens-backend/pull/11/files#diff-7c4bce5a12b1988cc43285bff091d0925cf01658cc8c5098a10f6d46f92b2b1eR30) them, however **these** tests are useful.

Reply:
okay, it's much clearer to me now) can you then please split this in two lines (with and without the protected method call) and maybe specify the noqa comment with the exact error code? the last part is very optional tho